code/mob/living/silicon/robot.dm

```diff
@@ -3240,6 +3240,48 @@
 /mob/living/silicon/robot/handle_event(var/event, var/sender)
 	hud.handle_event(event, sender)	// the HUD will handle icon_updated events, so proxy those
 
+/// Modify one tool in existing module, ex redeeming rewards or modififying sponge. Assumes item is in hand
+/mob/living/silicon/robot/proc/swap_individual_tool(var/obj/item/old_tool, var/obj/item/new_tool)
+	var/tool_index
+	var/tool_module_index
+
+	// Find index of the tool in hand
+	for (var/i = 1 to (src.module_states.len) step 1)
```

Comment:
```suggestion
	for (var/i = 1 to length(src.module_states))
```
`length()` is preferred, `step 1` is implied and doesn't really need to be written out